changelogs/fragments/9225-proxmox-module-refactoring.yml

```diff
@@ -0,0 +1,9 @@
+minor_changes:
+  - proxmox - refactors the proxmox module (https://github.com/ansible-collections/community.general/pull/9225)
+  - proxmox - fixes idempotency of template conversions (https://github.com/ansible-collections/community.general/pull/9225, https://github.com/ansible-collections/community.general/issues/8811).
+  - proxmox - fixes issues with disk_volume variable (https://github.com/ansible-collections/community.general/pull/9225, https://github.com/ansible-collections/community.general/issues/9065).
+  - proxmox - fixes incorrect parsing for bind-only mounts (https://github.com/ansible-collections/community.general/pull/9225, https://github.com/ansible-collections/community.general/issues/8982).
+  - proxmox module utils - fixes ignoring of ``choose_first_if_multiple`` argument in ``get_vmid`` (https://github.com/ansible-collections/community.general/pull/9225)
+
+deprecated_features:
+  - proxmox - removes default value of ``update`` parameter. This will be changed to a default of ``true`` in the future (https://github.com/ansible-collections/community.general/pull/9225)
```

Comment:
```suggestion
  - proxmox - deprecates default value ``false`` of ``update`` parameter. This will be changed to a default of ``true`` in community.general 11.0.0 (https://github.com/ansible-collections/community.general/pull/9225).
```